src/table.mjs

```diff
@@ -248,20 +248,44 @@ export function makeQueryTemplate(operations, source) {
   ];
   for (let i = 0; i < filter.length; ++i) {
     appendSql(i ? `\nAND ` : `\nWHERE `, args);
-    appendWhereEntry(filter[i], args);
+    appendWhereEntry(filter[i], args, escaper);
   }
   for (let i = 0; i < sort.length; ++i) {
     appendSql(i ? `, ` : `\nORDER BY `, args);
-    appendOrderBy(sort[i], args);
+    appendOrderBy(sort[i], args, escaper);
   }
-  if (slice.to !== null || slice.from !== null) {
-    appendSql(
-      `\nLIMIT ${slice.to !== null ? slice.to - (slice.from || 0) : 1e9}`,
-      args
-    );
-  }
-  if (slice.from !== null) {
-    appendSql(` OFFSET ${slice.from}`, args);
+  if (source.dialect === "mssql") {
+    if (slice.to !== null || slice.from !== null) {
+      if (!sort.length) {
+        if (columns[0] === "*")
```

Comment:
This isn’t how we represent SELECT *; when selecting all columns, `columns` will be null here (and so this will crash as currently written). See this test for example:

https://github.com/observablehq/stdlib/blob/f4a8b6de4a76d078063c8f4a0fe352aaf3781d3c/test/table-test.mjs#L50-L54

Reply:
Oh, you’re right; I was confusing `select.columns` (from the passed-in `operations`) and the local variable `columns`. You could use `columns === "*"` I suppose, then. But I think `!select.columns` would be more explicit and preferable.